### PR TITLE
SLVS-1258 Merge VS Name and Version logic into VsInfoProvider

### PR DESCRIPTION
**Pull request by vnaskos-sonar**

### CODE REVIEW

File: src/Infrastructure.VS/VsInfo/VsInfoProvider.cs [hunk 1]
```diff
@@ -75,5 +78,19 @@ private static IVsVersion CalculateVersion(IVsShell vsShell, ISetupConfiguration
...
+
+        private static string CalculateName(IVsShell vsShell)
+        {
+            const string defaultName = "Microsoft Visual Studio";
+            try
+            {
+                vsShell.GetProperty((int)__VSSPROPID5.VSSPROPID_AppBrandName, out var name);
+                return name as string ?? defaultName;
+            }
+            catch (Exception)
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Infrastructure.VS/VsInfo/VsInfoProvider.cs:90]: Should not catch critical exceptions

File: src/Integration.UnitTests/SLCore/SLCoreConstantsProviderTests.cs [hunk 2]
```diff
@@ -50,32 +48,6 @@ public void MefCtor_CheckIsSingleton()
         MefTestHelpers.CheckIsSingletonMefComponent<SLCoreConstantsProvider>();
     }
 
-    [TestMethod]
-    public void ClientConstants_ShouldBeExpected()
-    {
-        const string ideName = "MyIde";
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration.UnitTests/SLCore/SLCoreConstantsProviderTests.cs:56]: Is there no test for the ideName now?

**Review by georgii-borovinskikh-sonarsource — CHANGES_REQUESTED** [review]

File: src/Infrastructure.VS/VsInfo/VsInfoProvider.cs [hunk 3]
```diff
@@ -75,5 +78,19 @@ private static IVsVersion CalculateVersion(IVsShell vsShell, ISetupConfiguration
...
+
+        private static string CalculateName(IVsShell vsShell)
+        {
+            const string defaultName = "Microsoft Visual Studio";
+            try
+            {
+                vsShell.GetProperty((int)__VSSPROPID5.VSSPROPID_AppBrandName, out var name);
+                return name as string ?? defaultName;
+            }
+            catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Infrastructure.VS/VsInfo/VsInfoProvider.cs:90]: The same should actually be done for the other method

File: src/Infrastructure.VS.UnitTests/VsInfoProviderTests.cs [hunk 4]
```diff
@@ -127,21 +123,73 @@ public void Version_FailureToRetrieveSetupConfiguration_Null()
...
+            var vsShell = CreateVsShell();
+            var testSubject = CreateTestSubject(vsShell.Object);
+
+            var vsName = testSubject.Name;
+
+            vsName.Should().Be("Microsoft Visual Studio");
+        }
+        
+        [TestMethod]
+        public void Name_OnException_ReturnsDefault()
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Infrastructure.VS.UnitTests/VsInfoProviderTests.cs:175]: Test that it does not catch the CriticalException

**Review by georgii-borovinskikh-sonarsource — CHANGES_REQUESTED** [review]

File: src/Infrastructure.VS/VsInfo/VsInfoProvider.cs (hunk 3 above)
Comment (vnaskos-sonar) [reply src/Infrastructure.VS/VsInfo/VsInfoProvider.cs:90]: CriticalException is only relevant to Com calls

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

**Pull request merged**